### PR TITLE
Some small changes for thruster-socketio

### PR DESCRIPTION
**Pull request by rts-gordon**
hi @trezm 
  There are small changes for thruster-socketio:
 1, Change some log level to lower because there are so many logs; 
 2, Add a small function "get_rooms_number" in rooms module;
 3, Add some condition check in function "remove_socket_from_room" in rooms module.

Would you like to have a look at this PR? Thank you.

### CODE REVIEW

File: socketio_middleware/src/rooms.rs [hunk 1]
```diff
@@ -96,13 +100,24 @@ pub fn get_sockets_number_for_room(room_id: &str) -> usize {
 pub fn print_sockets_for_room(room_id: &str) {
     match ROOMS.get(room_id) {
         Some(sockets) => {
+            info!("ROOMS: room {} containt sockets number = {}.", room_id, sockets.len());
```
Comment (trezm) [inline socketio_middleware/src/rooms.rs]: Let's make this a `debug!`

File: socketio_middleware/src/rooms.rs [hunk 2]
```diff
@@ -96,13 +100,24 @@ pub fn get_sockets_number_for_room(room_id: &str) -> usize {
...
         }
 
         None => {
             debug!("ROOMS: no socket in room {}.", room_id);
         }
     }
 }
+
+///
+/// get rooms number
```
Comment (trezm) [inline socketio_middleware/src/rooms.rs]: A little more descriptive comment would be great, something like "Returns the number of rooms this middleware currently has reference too."

File: socketio_middleware/src/rooms.rs [hunk 3]
```diff
@@ -96,13 +100,24 @@ pub fn get_sockets_number_for_room(room_id: &str) -> usize {
...
+///
+pub fn get_rooms_number() -> usize {
```
Comment (trezm) [inline socketio_middleware/src/rooms.rs]: Let's change this to `get_room_count` instead, just a bit clearer.

File: socketio_middleware/src/socketio.rs [hunk 4]
```diff
@@ -53,7 +53,7 @@ pub async fn broadcast(room_id: &str, event: &str, message: &str) {
                         event.to_string(),
                         message.to_string(),
                     )));
-                    debug!("Found socketid {} in room {}, sending message = {}", channel.sid(), room_id, message);
+                    trace!("Found socketid {} in room {}, sending message = {}", channel.sid(), room_id, message);
```
Comment (trezm) [inline socketio_middleware/src/socketio.rs]: Keep as debug unless we have a rule as to when we use trace vs debug.

File: socketio_middleware/src/rooms.rs [hunk 5]
```diff
@@ -53,7 +53,7 @@ pub fn join_channel_to_room(room_id: &str, channel_pair: ChannelPair) {
     }
 
     if !exist {
-        debug!("ROOMS: socketid {} joined room {}.", channel_pair.sid(), room_id);
+        debug!("ROOMS: socketid {} joined room {}, room len = {}.", channel_pair.sid(), room_id, connected_sockets.len()+1);
```
Comment (trezm) [inline socketio_middleware/src/rooms.rs]: nit: add spacing, `connected_sockets.len() + 1`.

File: socketio_middleware/src/rooms.rs [hunk 6]
```diff
@@ -70,13 +70,17 @@ pub fn remove_socket_from_room(room_id: &str, sid: &str) {
...
-            debug!("ROOMS: socketid {} leave room {}.", socket.sid, room_id);
             connected_sockets.remove(i);
+            debug!("ROOMS: socketid {} leave room {}, room len = {}.", sid, room_id, connected_sockets.len());
             break;
         }
     }
 
-    ROOMS.insert(room_id.to_string(), connected_sockets);
+    //if there are still exist sockets, then insert back to ROOMS.
+    if connected_sockets.len() > 0 {
```
Comment (trezm) [inline socketio_middleware/src/rooms.rs:80]: This is a nice optimization! I like it!

**Review by trezm — CHANGES_REQUESTED** [review]

File: socketio_middleware/src/rooms.rs (hunk 1 above)
Comment (rts-gordon) [reply socketio_middleware/src/rooms.rs]: OK, changed to debug!

File: socketio_middleware/src/rooms.rs (hunk 3 above)
Comment (rts-gordon) [reply socketio_middleware/src/rooms.rs]: OK, changed to get_room_count

File: socketio_middleware/src/rooms.rs (hunk 2 above)
Comment (rts-gordon) [reply socketio_middleware/src/rooms.rs]: NP

File: socketio_middleware/src/socketio.rs (hunk 4 above)
Comment (rts-gordon) [reply socketio_middleware/src/socketio.rs]: NP

File: socketio_middleware/src/rooms.rs (hunk 5 above)
Comment (rts-gordon) [reply socketio_middleware/src/rooms.rs]: NP

File: socketio_middleware/src/rooms.rs (hunk 6 above)
Comment (rts-gordon) [reply socketio_middleware/src/rooms.rs:80]: sure, it is a better optimization.

**Comment by rts-gordon** [comment]
Hi @trezm 
  I have finished those new changes, can you please merge them and push to crates.io?  Thanks a lot.

**Review by trezm — APPROVED** [review]

**Pull request merged**